### PR TITLE
 libNX_X11 data files: Install libNX_X11 related data files.

### PR DESCRIPTION
**Pull request by sunweaver**
- Update XErrorDB to latest X.org version.
- Install XErrorDB, XKeysymDB and Xcms.txt to /usr/{local/}share/nx/.
- Assure that libNX_X11 finds those files at the
  named location.
- Update debian/ packaging files.
- Update nx-libs.spec packaging file.

### CODE REVIEW

**Comment by sunweaver** [comment]
@uli42: If you have any time over the weekend, I'd be happy to have this PR reviewed. I chose shipping XErrorDB in nx-libs for now, but have an updated copy from X.org.

File: nx-X11/config/cf/Imake.tmpl [hunk 1]
```diff
@@ -759,6 +759,25 @@ TCLIBDIR = TclLibDir
...
+#define AlternateUsrDataDir NO
+#endif
+#endif
+#else
+#ifndef AlternateUsrDataDir
+#define AlternateUsrDataDir YES
+#endif
+#endif
+
 #ifndef VarDirectory
```
Comment (uli42) [inline nx-X11/config/cf/Imake.tmpl:781]: I think some indention would help reading this.

```
#ifndef UsrDataDir
#  ifdef ProjectRoot
#    define UsrDataDir Concat4(ProjectRoot,/,share,/nx)
#    ifndef AlternateUsrDataDir
#      define AlternateUsrDataDir YES
#    endif
#  else
#    define UsrDataDir Concat4(/usr/local,/,share,/nx)
#    ifndef AlternateUsrDataDir
#      define AlternateUsrDataDir NO
#    endif
#  endif
#else
#  ifndef AlternateUsrDataDir
#    define AlternateUsrDataDir YES
#  endif
#endif
```

**Comment by uli42** [comment]
Apart from my one comment it looks good to me

File: nx-X11/config/cf/Imake.tmpl (hunk 1 above)
Comment (sunweaver) [inline nx-X11/config/cf/Imake.tmpl:781]: On  Mo 04 Jul 2016 22:40:16 CEST, Ulrich Sibiller wrote:

> > +#define UsrDataDir Concat4(ProjectRoot,/,share,/nx)
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir YES
> > +#endif
> > +#else
> > +#define UsrDataDir Concat4(/usr/local,/,share,/nx)
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir NO
> > +#endif
> > +#endif
> > +#else
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir YES
> > +#endif
> > +#endif
> > +
> >  #ifndef VarDirectory
> 
> I think some indention would help reading this.
> 
> ```
> #ifndef UsrDataDir
> #  ifdef ProjectRoot
> #    define UsrDataDir Concat4(ProjectRoot,/,share,/nx)
> #    ifndef AlternateUsrDataDir
> #      define AlternateUsrDataDir YES
> #    endif
> #  else
> #    define UsrDataDir Concat4(/usr/local,/,share,/nx)
> #    ifndef AlternateUsrDataDir
> #      define AlternateUsrDataDir NO
> #    endif
> #  endif
> #else
> #  ifndef AlternateUsrDataDir
> #    define AlternateUsrDataDir YES
> #  endif
> #endif
> ```

I immitated the style of the UsrLibDir definition further above in  
that file. I can add a commit after the one introducing UsrDataDir  
that does some indentation magic...

## Mike

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

Comment (sunweaver) [inline nx-X11/config/cf/Imake.tmpl:781]: On  Mo 04 Jul 2016 22:40:16 CEST, Ulrich Sibiller wrote:

> > +#define UsrDataDir Concat4(ProjectRoot,/,share,/nx)
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir YES
> > +#endif
> > +#else
> > +#define UsrDataDir Concat4(/usr/local,/,share,/nx)
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir NO
> > +#endif
> > +#endif
> > +#else
> > +#ifndef AlternateUsrDataDir
> > +#define AlternateUsrDataDir YES
> > +#endif
> > +#endif
> > +
> >  #ifndef VarDirectory
> 
> I think some indention would help reading this.
> 
> ```
> #ifndef UsrDataDir
> #  ifdef ProjectRoot
> #    define UsrDataDir Concat4(ProjectRoot,/,share,/nx)
> #    ifndef AlternateUsrDataDir
> #      define AlternateUsrDataDir YES
> #    endif
> #  else
> #    define UsrDataDir Concat4(/usr/local,/,share,/nx)
> #    ifndef AlternateUsrDataDir
> #      define AlternateUsrDataDir NO
> #    endif
> #  endif
> #else
> #  ifndef AlternateUsrDataDir
> #    define AlternateUsrDataDir YES
> #  endif
> #endif
> ```

Looked at Imake.tmpl once more. The whole file could need indentation,  
but that is really beyond the scope of this PR.

## Mike

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Pull request merged**